controllers/v1/observationsController.js

```diff
@@ -1794,6 +1795,91 @@ module.exports = class Observations extends Abstract {
                 });
             }
         });
+    }  
+
+      /**
+     * @api {post} /assessment/api/v1/observations/verifyLink Verify Observation Link And get details
+     * @apiVersion 1.0.0
+     * @apiName Verify Observation Link
+     * @apiGroup Observations
+     * @apiParamExample {json} Request-Body:
+     * {
+     *	    "data": {
+     *          "link":  "https://apps.shikshalokam.org/samiksha/create-observation/38cd93bdb87489c3890fe0ab00e7d406",
+     *      }
+     * }
+     * @apiUse successBody
+     * @apiUse errorBody
+     * @apiParamExample {json} Response:
+         {
+        "message": "Observation Link verified successfully",
+        "status": 200,
+        "result": [
+            {
+                "_id": "5f6853f293734140ccce90cf",
+                "entities": [
+                    "5f636fa2916c13367d8ff835"
+                ],
+                "createdFor": [],
+                "rootOrganisations": [],
+                "isAPrivateProgram": false,
+                "deleted": false,
+                "status": "published",
+                "solutionId": "5f64651f916c13367d8ff83f",
+                "solutionExternalId": "PRIYANKA-3-FRAMWORK-OBSERVATION-1",
+                "programId": "5f634e31577d2ce1ed942c65",
+                "programExternalId": "MY-ASSESSMENT-PROGRAM2",
+                "frameworkId": "5f6349c4577d2ce1ed942a56",
+                "frameworkExternalId": "PRIYANKA-3-FRAMWORK",
+                "entityTypeId": "5d15a959e9185967a6d5e8a6",
+                "entityType": "school",
+                "createdBy": "01c04166-a65e-4e92-a87b-a9e4194e771d",
+                "startDate": "2020-09-21T07:19:14.618Z",
+                "endDate": "2021-09-21T07:19:14.618Z",
+                "name": "Priyanka Observation solution",
+                "description": "Priyanka Observation description",
+                "updatedAt": "2020-09-21T07:19:14.648Z",
+                "createdAt": "2020-09-21T07:19:14.648Z",
+                "__v": 0,
+                "link": "a325411f49158bc21b7f08d33aad5c02"
+            }
+        ]
+    }
+    */
+     
+    /**
+    * Verify Observation Link.
+    * @method
+    * @name verifyLink
+    * @param {Object} req -request Data.
+    * @returns {JSON} - bservation data.
+    */
+
+    verifyLink(req) {
+        return new Promise(async (resolve, reject) => {
+ 
+            try {
+
+                let result = await observationsHelper.verify(
+                    req.body.data,
```

Comment:
@priyanka-TL  If data is empty or link is not provided you need to make a check in validator function inside solutions module.